File: src/main/scala/com/jjw/spark/test1/Operator_foreach.scala
package com.jjw.spark.test1

import org.apache.spark.SparkConf
import org.apache.spark.SparkContext

object Operator_foreach {
  def main(args: Array[String]): Unit = {
    val conf = new SparkConf()
    conf.setMaster("local").setAppName("foreach")
    val sc = new SparkContext(conf)
    val rdd1 = sc.parallelize(Array(1,2,3,4))
    rdd1.foreach { println }
    
    sc.stop()
  }
}

File: src/main/java/com/jjw/sparkCore/actions/Operator_foreach.java
package com.jjw.sparkCore.actions;

import java.util.Arrays;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.VoidFunction;

public class Operator_foreach {
	public static void main(String[] args) {
		SparkConf conf = new SparkConf();
		conf.setMaster("local").setAppName("foreach");
		JavaSparkContext sc = new JavaSparkContext(conf);
		JavaRDD<Integer> parallelize = sc.parallelize(Arrays.asList(1,2,3));
		parallelize.foreach(new VoidFunction<Integer>() {
			
			/**
			 * 
			 */
			private static final long serialVersionUID = 1L;

			@Override
			public void call(Integer t) throws Exception {
				System.out.println(t);
			}
		});
		
		sc.stop();
	}
}
